lib/manageiq/api/common/entitlement.rb

```diff
@@ -0,0 +1,35 @@
+module ManageIQ
+  module API
+    module Common
+      class EntitlementError < StandardError; end
+
+      class Entitlement
+        def initialize(identity)
+          @identity = identity
+        end
+
+        %w[
+          hybrid_cloud
+          insights
+          openshift
+          smart_management
+        ].each do |m|
```

Comment:
why is this list hardcoded ? I'd expect this to be dynamic based on the identity.entitlements content, minimizing churn in this code as the entitlement service adds things in the future. The dynamic methods can be created during initialization based on the identity.

Reply:
Hi @abellotti, it's hardcoded per the request by Jason to do the same on https://github.com/ManageIQ/manageiq-api-common/blob/master/lib/manageiq/api/common/user.rb#L11-L24

My thought was the churn here will need to be managed anyway. So controlling it instead of new elements showing up unannounced makes sense.

Reply:
I also don't think we should hardcode this, as it's likely to change frequently, unlike the identity keys, which are likely static.

That being said, I'm ok with merging for now, but I think this particular bit should be dynamic.

Reply:
I agree with @syncrou's thought. Explicitly managing the elements make the code easy to read and use. Do we really expect major products are added to the list frequently?